cdap-ui/app/cdap/components/SpotlightSearch/index.js

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {Component} from 'react';
+import classnames from 'classnames';
+import {MySearchApi} from 'api/search';
+import {default as NamespaceStore} from 'services/store/store';
+import {parseMetadata} from 'services/metadata-parser';
+import { Dropdown, DropdownMenu, DropdownItem } from 'reactstrap';
+import debounce from 'lodash/debounce';
+import SpotlightModal from 'components/SpotlightSearch/SpotlightModal';
+import Mousetrap from 'mousetrap';
+import T from 'i18n-react';
+require('./SpotlightSearch.less');
+
+const keyMap = {
+  enter: 13,
+  esc: 27,
+  arrowUp: 38,
+  arrowDown: 40
+};
+
+const VIEW_RESULT_LIMIT = 5;
+
+export default class SpotlightSearch extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSearchClick = this.handleSearchClick.bind(this);
+    this.debounceSearch = debounce(this.handleSearch.bind(this, event), 300);
```

Comment:
Now that we are using RxJs streams and since we have a ref for the search textbox can we use one stream to listen to changes to the searchbox as we have debounce operator in rxjs. Would be interesting to find what works best for us. Is it `onChange` of react or is it rxjs stream attached to the event.